lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -174,7 +174,10 @@ def relationship_value_for(serializer, options = {})
         end
 
         def relationships_for(serializer)
-          serializer.associations.each_with_object({}) do |association, hash|
+          resource_type = resource_identifier_type_for(serializer)
+          requested_associations = fieldset.try(:fields_for, resource_type) || '*'
```

Comment:
I believe code should be clear and self-explanatory. If a new contributor would be confused, then I'd rather change the code to something clearer than add comments. ["Good code is its own best documentation"](https://github.com/bbatsov/ruby-style-guide#no-comments)